fabric-item-api-v1/src/main/java/net/fabricmc/fabric/api/item/v1/FabricItemSettings.java

```diff
@@ -32,6 +32,42 @@
  * {@code new FabricItemSettings()}.
  */
 public class FabricItemSettings extends Item.Settings {
+	public enum AttackSound {
+		STRONG,
+		WEAK,
+		CRITIAL,
+		KNOCKBACK,
+		NO_DAMAGE,
+		SWEEPING,
+	}
+
+	public FabricItemSettings soundEvent(AttackSound sound, SoundPlayer soundEvent) {
+		switch (sound) {
+		case CRITIAL:
+			FabricItemInternals.computeExtraData(this).criticalHitSound(soundEvent);
+			break;
+		case KNOCKBACK:
+			FabricItemInternals.computeExtraData(this).knockbackHitSound(soundEvent);
+			break;
+		case NO_DAMAGE:
+			FabricItemInternals.computeExtraData(this).noDamageHitSound(soundEvent);
+			break;
+		case STRONG:
+			FabricItemInternals.computeExtraData(this).strongHitSound(soundEvent);
+			break;
+		case SWEEPING:
+			FabricItemInternals.computeExtraData(this).sweepingHitSound(soundEvent);
+			break;
+		case WEAK:
+			FabricItemInternals.computeExtraData(this).weakHitSound(soundEvent);
+			break;
+		default:
+			break;
```

Comment:
What should we do in the `default` case?

Reply:
Is it bad form to just remove it?

Reply:
Well removing the break case would essentially be a "do nothing". If the compiler requires it then it would be empty if we don't want to fail.